### DISCUSSION TITLE
Allow stop without previous start command

### DISCUSSION THREAD

**Discussion by rkj** [jotaen/klog#362] [Help / Question]
I frequently forget to run start command, and then need to look up when was end time of my previous entry and use `track` with full times to create it. It would be nice if I could just use it like:
```bash
klog start # only used once in the morning
klog stop -s 'Fixed bug X'
klog stop -s 'Reviewed doc Y' # takes start time from the end of previous activity
```

**Comment by rkj** [comment]
I've found simple bash workaround:
```
klog_stop() {
  klog start --time $(klog json | jq '.records[-1].entries[-1].end' -r)
  klog stop -s "'$*'"
}
```

**Comment by jotaen** [comment]
Retroactively starting a time range is tricky, because it also might be undesired, or cover up a mistake. It’s probably better to make the behaviour explicit, as with your workaround for example.

As alternative idea to your solution, I could also imagine a custom command like `klog_checkin` or `klog_commit`, which does `klog stop` and `klog start` under the hood. So you’d only do `klog start` at the very beginning, and then you’d use `klog_commit` throughout the day, to continuously flush the individual activities.

```bash
klog_commit() {
  klog stop -s "'$*'"
  klog start
}
```
